environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentStopService.java

```diff
@@ -80,4 +86,17 @@ private void validaFreeIpaIsNotRunningOnSpotInstances(EnvironmentDto environment
             throw new BadRequestException(message);
         }
     }
+
+    private void validateAttachedSDXClustersCanBeStopped(EnvironmentDto environment) {
+        List<SdxClusterResponse> attachedSdxClusters = sdxService.list(environment.getName());
+        for (SdxClusterResponse response : attachedSdxClusters) {
+            if (!sdxService.isStoppable(response.getCrn())) {
+                String message = String.format(
+                        "Environment [%s] can not be stopped because SDX cluster [%s] currently can not be stopped.",
```

Comment:
If datalake service would return a description why it can't be stopped then we could include this in the message.